добавил проверку результатов

good5874 opened the pull request.

good5874 committed:
добавил проверку результатов

good5874 committed:
добавил сохранение результатов теста, можно помотреть всю историю пройденых тестов или по названию отдельного теста.

BitW13 commented on EpamTestConsole/Checking/CheckedQuestion.cs:
```diff
@@ -0,0 +1,23 @@
+        public string Answer { get; set; }
+        public string Result { get; set; }
+
+        public CheckedQuestion(string _NameSection, string _TextQuestion,string _Answer, string _Result)
```
без нижних подчеркиваний и с нижнего регистра

BitW13 commented on EpamTestConsole/Checking/CheckedQuestions.cs:
```diff
@@ -0,0 +1,73 @@
+namespace EpamTestConsole
+{
+    [Serializable]
+    public class CheckedQuestions
```
Не очень хорошее название класса с учетом того, что у тебя есть CheckedQuestion

BitW13 commented on EpamTestConsole/Checking/CheckedQuestions.cs:
```diff
@@ -0,0 +1,73 @@
+    [Serializable]
+    public class CheckedQuestions
+    {
+        // 4 типа вопросов проверет:
```
Лучше тогда разбить проверку на несколько методов. по методу на тип

BitW13 commented on EpamTestConsole/Checking/CheckedQuestions.cs:
```diff
@@ -0,0 +1,73 @@
+        // 3. вопрос где сравнивается введённая строка пользователем с ответом на вопрос в тесте
+        // 4. вопрос где сравнивается выбранный вариант ответа с ответом на вопрос в тесте
+
+        public List<CheckedQuestion> VerifiedQuestions { get; private set; } = new List<CheckedQuestion>();
```
И почему ты разделяешь Question от CheckedQuestion? Разве это разные сущности?
Подумай над тем, как можно их совместить

BitW13 commented on EpamTestConsole/Checking/CheckedQuestions.cs:
```diff
@@ -0,0 +1,73 @@
+                        VerifiedQuestions.Add
+                            (
+                            new CheckedQuestion(section.NameSection, section.Questions[i].TextQuestion, answers[i],
+                            CheckingQuestion(section.Questions[i], answers[i]).ToString()
```
Question, CheckedQuestion и CheckingQuestion..... Это говорит о том, что что-то пошло не так)

good5874 replied:
Скорее всего))))

BitW13 commented on EpamTestConsole/Console/Menu.cs:
```diff
@@ -294,30 +315,89 @@ private void AddSection(TreeNode node)
-                Console.WriteLine("Введите название подтем:");
-                for (int i = 1; i < sectionCount + 1; i++)
+
+                string strsectionCount = Console.ReadLine();
```
strsectionCount?

good5874 replied:
;)

good5874 committed:
Поправивил Question, переопределил ToString, добавил в него проверку ответов
В menu теперь создаёт объекты question и section, а management добавляет их в нужный узел.

good5874 committed:
исправил опечатку

good5874 committed:
добавил проверку вопроса и вывод в консоль сразу после ответа на вопрос

good5874 committed:
Timer
